lib/worker.js

```diff
@@ -259,23 +287,51 @@ class AssetComputeWorker {
 
         // post-process and upload
         for (let rendition of this.renditions) {
+            setConsoleLogPrefix(`[rendition ${rendition.index}]`);
+            try {
 
-            rendition = await this.postProcess(rendition);
-            if (!rendition) {
-                continue;
-            }
+                rendition = await this.postProcess(rendition);
+                if (!rendition) {
+                    continue;
+                }
 
-            if (this.options.disableRenditionUpload) {
-                await this.renditionSuccess(rendition);
-            } else if (rendition.exists()) {
-                await this.upload(rendition);
-            } else {
-                console.error(`no rendition found at: ${rendition.path}`);
-                await this.renditionFailure(rendition, new GenericError(`No rendition generated for ${rendition.id()}`, `${this.actionName}_batchProcess_norendition`));
+                if (this.options.disableRenditionUpload) {
+                    await this.renditionSuccess(rendition);
+                } else if (rendition.exists()) {
+                    await this.upload(rendition);
+                } else {
+                    console.error(`no rendition found at: ${rendition.path}`);
+                    await this.renditionFailure(rendition, new GenericError(`No rendition generated for ${rendition.id()}`, `${this.actionName}_batchProcess_norendition`));
+                }
+            } finally {
+                setConsoleLogPrefix();
             }
         }
     }
 
+    async preparePostProcess(rendition) {
+        // post processing could ask for different instructions for the worker callback
+        const intermediateInstructions = await prepareImagePostProcess(rendition, this.options);
+        if (intermediateInstructions) {
+            console.log("original instructions:", rendition.instructionsForEvent());
+
+            rendition.changeInstructions(intermediateInstructions);
```

Comment:
I think it would be useful to log the updated instructions too here, so they're easy to compare.
```suggestion
            rendition.changeInstructions(intermediateInstructions);
            console.log("updated instructions:", rendition.instructionsForEvent());
```

Reply:
They are logged [later already](https://github.com/adobe/asset-compute-sdk/blob/b57ebd3506b5015166d9ba2e5e792dadcf2972c5/lib/worker.js#L207). I had that in here, but then it was just bloating especially if the instructions are larger (and include the `userData` from AEM or IT tests) and you see them basically twice in succession.

Reply:
Ack.